### PR TITLE
[rush-lib] Fix the issue where update-autoinstaller does not use .npmrc

### LINKED ISSUE #2905: [rush] update-autoinstaller does not use registry in npmrc
rush update-autoinstaller --name xxx

## Summary

update-autoinstaller via `pnpm` does not use registry in `common/config/rush/.npmrc`

## Repro steps

  **Expected result:** 

Rush uses corresponding registry defined in npmrc

  **Actual result:** 

Failed to fetch certain library which is not available in public npm registry

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? |  5.47.0 |
| `rushVersion` from rush.json? |  5.47.0 |
| `useWorkspaces` from rush.json? |  true |
| Operating system? |  MacOS |
| Would you consider contributing a PR? |  Yes |
| Node.js version (`node -v`)? |  14.16.1 |

### PR DESCRIPTION
**Pull request by h-a-n-a**
## Summary

Fixes #2905 

## Details

Copy `.npmrc` to the current working autoinstaller
## How it was tested

Reinvoked `rush update-autoinstaller --name xxx` and all the dependencies installed correctly that private registry provides.

### CODE REVIEW

**Comment by D4N14L** [comment]
cc @octogonz who added this feature.

File: apps/rush-lib/src/logic/Autoinstaller.ts [hunk 1]
```diff
@@ -80,13 +80,17 @@ export class Autoinstaller {
...
+    Utilities.syncNpmrc(this._rushConfiguration.commonRushConfigFolder, this.folderFullPath);
+
     Utilities.executeCommand({
       command: this._rushConfiguration.packageManagerToolFilename,
       args: ['install'],
       workingDirectory: this.folderFullPath,
       keepEnvironment: true
     });
 
+    FileSystem.deleteFile(path.join(this.folderFullPath, '/.npmrc'));
```
Comment (D4N14L) [inline apps/rush-lib/src/logic/Autoinstaller.ts:92]: Why delete after? Given that we sync the most current NPMRC (or delete if the current NPMRC isn't found) before running the command, would the delete be necessary?

Comment (h-a-n-a) [reply apps/rush-lib/src/logic/Autoinstaller.ts:92]: I was thinking if rush should preserve user's file structure, but I just found rush project template ignored the file. So I will remove this.

Comment (h-a-n-a) [reply apps/rush-lib/src/logic/Autoinstaller.ts]: Done.

File: common/changes/@microsoft/rush/fix-autoinstaller-update_2021-09-14-17-35.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "changes": [
+    {
+      "packageName": "@microsoft/rush",
+      "comment": "fix the issue where update-autoinstaller does not use .npmrc",
```
Comment (iclanton) [inline common/changes/@microsoft/rush/fix-autoinstaller-update_2021-09-14-17-35.json:5]:
```suggestion
      "comment": "Fix an issue where `rush update-autoinstaller` does not use the repo's .npmrc",
```

**Review by iclanton — APPROVED** [review]

**Pull request merged**